clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -288,6 +291,20 @@ private void throwIfOffsetOutOfRange() throws OffsetOutOfRangeException {
     }
 
     /**
+     * If any topic from previous fetchResponse contatains Authorization error, throw ApiException.
+     * @throws ApiException
+     */
+    private void throwIfUnauthorized() throws ApiException {
+        if (!unauthorizedTopicPartitions.isEmpty()) {
+            StringBuilder sb = new StringBuilder();
+            for (TopicPartition topicPartition : unauthorizedTopicPartitions)
+                sb.append(topicPartition + ",");
+            unauthorizedTopicPartitions.clear();
+            throw new ApiException(String.format("Not authorized to read from %s", sb.substring(0, sb.length() - 1).toString()));
```

Comment:
Perhaps it's better to have an explicit AuthorizationException and just throw that.